src/Semantic/Api/Bridge.hs

```diff
@@ -71,8 +78,18 @@ instance APIBridge T.Text Data.Language where
 
 instance APIBridge API.Blob Data.Blob where
   bridging = iso apiBlobToBlob blobToApiBlob where
-    blobToApiBlob b = defMessage & P.content .~ toText (Data.blobSource b) & P.path .~ T.pack (Data.blobPath b) & P.language .~ (bridging # Data.blobLanguage b)
-    apiBlobToBlob blob = Data.makeBlob (fromText (blob^.content)) (T.unpack (blob^.path)) (blob^.(language . bridging)) mempty
+    blobToApiBlob b
+      = defMessage
+      & P.content .~ Source.toText (Data.blobSource b)
+      & P.path .~ T.pack (Data.blobPath b)
+      & P.language .~ (bridging # Data.blobLanguage b)
+    apiBlobToBlob blob =
+      let src = blob^.content.to Source.fromText
+          pth = fromRight (Path.toAbsRel Path.emptyFile) (blob^.path._Text.to Path.parse)
```

Comment:
This gyration is to handle the degenerate case of if we get an invalid path over the wire. This could conceivably violate the `Iso` laws but I truly do not find this important.

Reply:
Yeah, that seems reasonable 👍